agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -1776,6 +1779,24 @@ bool backhaul_manager::handle_1905_topology_query(ieee1905_1::CmduMessageRx &cmd
 
     tlvDeviceInformation->mac() = network_utils::mac_from_string(bridge_info.mac);
 
+    uint32_t speed;
+    if (network_utils::linux_iface_get_speed(bridge_info.iface, speed)) {
+        std::shared_ptr<ieee1905_1::cLocalInterfaceInfo> localInterfaceInfo =
+            tlvDeviceInformation->create_local_interface_list();
+
+        ieee1905_1::eMediaType media_type = ieee1905_1::eMediaType::UNKNONWN_MEDIA;
```

Comment:
I think all this can go to a static function:
```
static bool fill_iface_info(ieee1905_1::tlvDeviceInformation &info, const std::string iface, const sMacAddr &iface_mac);
```

Then call it - `fill_iface_info(*info, bridge_info.iface, network_utils::mac_from_string(bridge_info.mac));`

Reply:
We still have to add a LocalInterfaceInfo field for each wireless interface. I'll refactor this code when I had the full picture. Promise.